azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/runtime/DockerFunctionRuntimeHandler.java

```diff
@@ -0,0 +1,106 @@
+
+import static com.microsoft.azure.maven.Utils.assureServerExist;
+import static com.microsoft.azure.maven.appservice.DockerImageType.PUBLIC_DOCKER_HUB;
+import static com.microsoft.azure.maven.function.Constants.APP_SETTING_FUNCTION_APP_EDIT_MODE;
+import static com.microsoft.azure.maven.function.Constants.APP_SETTING_FUNCTION_APP_EDIT_MODE_VALUE;
+import static com.microsoft.azure.maven.function.Constants.APP_SETTING_MACHINEKEY_DECRYPTION_KEY;
+import static com.microsoft.azure.maven.function.Constants.APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE;
+import static com.microsoft.azure.maven.function.Constants.APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE_VALUE;
+
+public class DockerFunctionRuntimeHandler extends AbstractLinuxFunctionRuntimeHandler {
+
+    private static final String INVALID_DOCKER_RUNTIME = "Invalid docker runtime configured.";
+
+    public static class Builder extends FunctionRuntimeHandler.Builder<DockerFunctionRuntimeHandler.Builder> {
+
+        @Override
+        public DockerFunctionRuntimeHandler build() {
+            return new DockerFunctionRuntimeHandler(self());
+        }
+
+        @Override
+        protected Builder self() {
+            return this;
+        }
+    }
+
+    protected DockerFunctionRuntimeHandler(Builder builder) {
+        super(builder);
+    }
+
+    @Override
+    public FunctionApp.DefinitionStages.WithCreate defineAppWithRuntime() throws MojoExecutionException {
+        final Server server = Utils.getServer(settings, serverId);
+        final DockerImageType imageType = AppServiceUtils.getDockerImageType(image, serverId, registryUrl);
+        checkFunctionExtensionVersion();
+        checkServerConfiguration(imageType, server);
+
+        final FunctionApp.DefinitionStages.WithDockerContainerImage withDockerContainerImage = super.defineLinuxFunction();
+        final FunctionApp.DefinitionStages.WithCreate result;
+        switch (imageType) {
+            case PUBLIC_DOCKER_HUB:
+                result = withDockerContainerImage.withPublicDockerHubImage(image);
+                break;
+            case PRIVATE_DOCKER_HUB:
+                result = withDockerContainerImage.withPrivateDockerHubImage(image).withCredentials(server.getUsername(), server.getPassword());
+                break;
+            case PRIVATE_REGISTRY:
+                result = withDockerContainerImage.withPrivateRegistryImage(image, registryUrl).withCredentials(server.getUsername(), server.getPassword());
+                break;
+            default:
+                throw new MojoExecutionException(INVALID_DOCKER_RUNTIME);
+        }
+        final String decryptionKey = generateDecryptionKey();
+        return (FunctionApp.DefinitionStages.WithCreate) result
+                .withAppSetting(APP_SETTING_MACHINEKEY_DECRYPTION_KEY, decryptionKey)
+                .withAppSetting(APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE, APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE_VALUE)
```

Comment:
How can a code reader find APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE_VALUE = false?

Reply:
Same as APP_SETTING_FUNCTION_APP_EDIT_MODE_VALUE

Reply:
Will fix naming in later pr